experiments/PlaywrightSharp/PlaywrightSharp/ApiSurface/IBrowserType.cs

```diff
@@ -0,0 +1,83 @@
+	/// example of using Playwright to drive automation:
+	/// </summary>
+	public interface IBrowserType
+	{
+		/// <summary>
+		/// A path where Playwright expects to find a bundled browser executable.
+		/// </summary>
+		string GetExecutablePath();
+		/// <summary>
+		/// Returns the browser instance.
+		/// You can use {OPTION} to filter out `--mute-audio` from default arguments:
+		/// > **Chromium-only** Playwright can also be used to control the Chrome browser, but it works best with the version of Chromium
+		/// it is bundled with. There is no guarantee it will work with any other version. Use {OPTION} option with extreme caution.
+		/// >
+		/// > If Google Chrome (rather than Chromium) is preferred, a <a href="https://www.google.com/chrome/browser/canary.html">Chrome Canary</a> or
+		/// <a href="https://www.chromium.org/getting-involved/dev-channel">Dev Channel</a> build is suggested.
+		/// >
+		/// > In <see cref="IBrowserType.LaunchAsync"/> above, any mention of Chromium also applies to Chrome.
+		/// >
+		/// > See <a href="https://www.howtogeek.com/202825/what%E2%80%99s-the-difference-between-chromium-and-chrome/">`this article`</a> for
+		/// a description of the differences between Chromium and Chrome. <a href="https://chromium.googlesource.com/chromium/src/+/lkgr/docs/chromium_browser_vs_google_chrome.md">`This article`</a> describes
+		/// some differences for Linux users.
+		/// </summary>
+		Task<IBrowser> LaunchAsync(string[] args, bool chromiumSandbox, bool devtools, path downloadsPath, IEnumerable<KeyValuePair<string, string>> env, path executablePath, IEnumerable<KeyValuePair<string, string>> firefoxUserPrefs, bool handleSIGHUP, bool handleSIGINT, bool handleSIGTERM, bool headless, bool ignoreDefaultArgs, string[] ignoreDefaultArgsValues, object proxy, float slowMo, float timeout);
+		/// <summary>
+		/// Returns the persistent browser context instance.
+		/// Launches browser that uses persistent storage located at {PARAM} and returns the only context. Closing this context will
+		/// automatically close the browser.
+		/// </summary>
+		Task<IBrowserContext> LaunchPersistentContextAsync(path userDataDir, bool acceptDownloads, string[] args, bool bypassCSP, bool chromiumSandbox, ColorScheme colorScheme, float deviceScaleFactor, bool devtools, path downloadsPath, IEnumerable<KeyValuePair<string, string>> env, path executablePath, IEnumerable<KeyValuePair<string, string>> extraHTTPHeaders, object geolocation, bool handleSIGHUP, bool handleSIGINT, bool handleSIGTERM, bool hasTouch, bool headless, object httpCredentials, bool ignoreDefaultArgs, string[] ignoreDefaultArgsValues, bool ignoreHTTPSErrors, bool isMobile, bool javaScriptEnabled, string locale, bool offline, string[] permissions, object proxy, float slowMo, float timeout, string timezoneId, string userAgent);
+		/// <summary>
+		/// Returns browser name. For example: `'chromium'`, `'webkit'` or `'firefox'`.
+		/// </summary>
+		string GetName();
```

Comment:
I think we talk about this. I think that non-async methods like this should be resolve to Properties.